packages/babel-core/src/transformation/file/options/option-manager.js

```diff
@@ -282,16 +282,27 @@ export default class OptionManager {
    */
   resolvePresets(presets: Array<string | Object>, dirname: string, onResolve?) {
     return presets.map(val => {
+      let options;
+      if (Array.isArray(val)){
+        [val, options] = val;
```

Comment:
IMO code will be more clear and readable if you rename `val` to `presetName` in left side of destructuring assignment.